src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -2912,33 +2954,42 @@ public void resetLatestFlagAsFalse(List<ADTask> adTasks) {
         if (adTasks == null || adTasks.size() == 0) {
             return;
         }
-        BulkRequest bulkRequest = new BulkRequest();
+        List<BulkOperation> operations = new ArrayList<>();
         adTasks.forEach(task -> {
             try {
+
                 task.setLatest(false);
                 task.setLastUpdateTime(Instant.now());
-                IndexRequest indexRequest = new IndexRequest(DETECTION_STATE_INDEX)
-                    .id(task.getTaskId())
-                    .source(task.toXContent(XContentBuilder.builder(XContentType.JSON.xContent()), XCONTENT_WITH_TYPE));
-                bulkRequest.add(indexRequest);
+
+                BulkOperation op = new BulkOperation.Builder()
+                    .index(b -> b.index(DETECTION_STATE_INDEX).id(task.getTaskId()).document(task))
```

Comment:
Another benefit of using JavaClient. No need to parse XContent. We can parse the model object directly!